File: Java/src/test/java/biz/shop/RabattrechnerTest.java
package biz.shop;

import static org.junit.Assert.assertThat;

import java.math.BigDecimal;

import org.hamcrest.Matcher;
import org.junit.Test;

class RabattrechnerTest {

    private Rabattrechner rechner = new Rabattrechner();

    @Test
    void shouldGiveNoRabattForPurchaseBelow1000Eur() {
        BigDecimal purchaseBelow1000Eur = toMoney(275);

        BigDecimal rabatt = rechner.calcRabattFor(purchaseBelow1000Eur);

        assertThat(rabatt, is(0));
    }

    @Test
    void shouldGiveNoRabattForPurchaseJust1000Eur() {
        BigDecimal purchaseBelow1000Eur = toMoney(999);

        BigDecimal rabatt = rechner.calcRabattFor(purchaseBelow1000Eur);

        assertThat(rabatt, is(0));
    }

    @Test
    void shouldGive3PercentRabattForPurchaseAbove1000Eur() {
        BigDecimal purchaseAbove1000Eur = toMoney(2000);

        BigDecimal rabatt = rechner.calcRabattFor(purchaseAbove1000Eur);

        assertThat(rabatt, is(60));
    }

    @Test
    void shouldGive3PercentRabattForPurchaseOfExactly1000Eur() {
        BigDecimal purchaseOfExactly1000Eur = toMoney(1000);

        BigDecimal rabatt = rechner.calcRabattFor(purchaseOfExactly1000Eur);

        assertThat(rabatt, is(30));
    }

    private static BigDecimal toMoney(int euros) {
        return new BigDecimal(euros);
    }

    private Matcher<? super BigDecimal> is(int euros) {
        return BigDecimalMatchers.is(toMoney(euros));
    }

}

File: PHP/test/RabattRechnerTest.php
<?php

namespace Test;

use Brick\Math\BigDecimal;
use PHPUnit\Framework\TestCase;
use Shop\RabattRechner;

class RabattRechnerTest extends TestCase
{

    private $calculator;

    /** @before */
    public function setUpDiscount()
    {
        $this->calculator = new RabattRechner();
        $this->calculator->addDiscountLevel(BigDecimal::of(1000), 3);
    }

    /** test */
    public function foo()
    {
        $this->assertEquals(true, false);
    }

    /** @test */
    public function shouldGiveNoRabattForPurchaseBelow1000Eur()
    {
        $purchaseBelow1000Eur = BigDecimal::of(275);
        $rabatt = $this->calculator->discountFor($purchaseBelow1000Eur);
        $this->assertEquals(BigDecimal::of(0), $rabatt);
    }

    /** @test */
    public function shouldGiveNoRabattForPurchaseJust1000Eur()
    {
        $purchaseBelow1000Eur = BigDecimal::of(999);
        $rabatt = $this->calculator->discountFor($purchaseBelow1000Eur);
        $this->assertEquals(BigDecimal::of(0), $rabatt);
    }

    /** @test */
    public function shouldGive3PercentRabattForPurchaseAbove1000Eur()
    {
        $purchaseAbove1000Eur = BigDecimal::of(2000);
        $rabatt = $this->calculator->discountFor($purchaseAbove1000Eur);
        $this->assertTrue(BigDecimal::of(60)->isEqualTo($rabatt));
    }

    /** @test */
    public function shouldGive3PercentRabattForPurchaseOfExactly1000Eur()
    {
        $purchaseOfExactly1000Eur = BigDecimal::of(1000);
        $rabatt = $this->calculator->discountFor($purchaseOfExactly1000Eur);
        $this->assertTrue(BigDecimal::of(30)->isEqualTo($rabatt));
    }

}
